### PR TITLE
Istio update to use proxy image with path normalization

### PR DESCRIPTION
**Pull request by jrajahalme**
Update Istio GSG and CI tests to use Cilium versions of Istio proxy (1.1.3) and pilot (1.1.2).

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7712)

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
Tests succeeded but ``Archive the artifacts - (6 min 15 sec in self)`` timed out

**Comment by jrajahalme** [comment]
Updated the proxy to 1.1.3, which is built with path normalization.

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

File: Documentation/gettingstarted/istio.rst [hunk 1]
```diff
@@ -99,6 +97,18 @@ Deploy Istio onto Kubernetes:
 ::
 
     $ kubectl create namespace istio-system
+    $ helm template ${ISTIO_HOME}/install/kubernetes/helm/istio-init --name istio-init --namespace istio-system | kubectl apply -f -
+
+Verify that 53 Istio CRDs have been created:
+
+::
+
+    $ kubectl get crds | grep 'istio.io\|certmanager.k8s.io' | wc -l
```
Comment (ianvernon) [inline Documentation/gettingstarted/istio.rst]: Would be nice to have something like `watch -n 1 "kubectl get crds | grep 'istio.io\|certmanager.k8s.io' | wc -l"` instead so users don't have to keep entering the same command again and again.

**Review by ianvernon — APPROVED** [review]

**Comment by ianvernon** [comment]
Backporting the CI changes may prove difficult.

**Comment by jrajahalme** [comment]
#7615

**Comment by jrajahalme** [comment]
test-me-please

File: Documentation/gettingstarted/istio.rst (hunk 1 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst]: I'll try that and update accordingly!

**Comment by jrajahalme** [comment]
@ianvernon Maybe it's good enough if the master (and maybe 1.5) has the improved CI, and just backport the fix itself to earlier versions?

**Comment by jrajahalme** [comment]
I'll split up the CI changes to a separate commit to make selective backporting easier.

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
@ianvernon Do you know if it is possible to re-run the Travis (unit) tests without triggering another Ginkgo run?

File: Documentation/gettingstarted/istio.rst [hunk 2]
```diff
@@ -71,21 +81,20 @@ Cilium's Envoy filters to query the Cilium agent for policy configuration:
...
 ::
 
     $ helm template istio-cilium-helm --name istio --namespace istio-system \
-          --set pilot.image=docker.io/cilium/istio_pilot:1.0.2 \
+          --set pilot.image=docker.io/cilium/istio_pilot:${ISTIO_VERSION} \
           --set sidecarInjectorWebhook.enabled=false \
           --set global.controlPlaneSecurityEnabled=true \
           --set global.mtls.enabled=true \
-          --set global.proxy.image=proxy_debug \
+          --set global.proxy.image=docker.io/cilium/istio_proxy:1.1.3 \
```
Comment (nebril) [inline Documentation/gettingstarted/istio.rst:97]: is this supposed to be `1.1.3` and not `1.1.2`?

File: test/k8sT/istio.go [hunk 3]
```diff
@@ -90,13 +93,22 @@ var _ = Describe("K8sIstioTest", func() {
 		res.ExpectSuccess("unable to create namespace %q", istioSystemNamespace)
 
 		By("Creating the Istio resources")
+
+		res = kubectl.Apply(istioCRDYAMLPath)
+		res.ExpectSuccess("unable to create Istio CRDs")
+
+		By("Waiting for Istio CRDs to be ready")
+		err := kubectl.WaitForCRDCount("istio.io|certmanager.k8s.io", 53, helpers.HelperTimeout)
```
Comment (nebril) [inline test/k8sT/istio.go:101]: 53 looks really magical here, can we move it out to a variable with descriptive name?

**Review by nebril — COMMENTED** [review]
Two nitpicks left in comments, but not really critical, please fix if you think it's worth it.

File: Documentation/gettingstarted/istio.rst (hunk 2 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst:97]: 1.1.3 is correct here.

File: test/k8sT/istio.go (hunk 3 above)
Comment (jrajahalme) [reply test/k8sT/istio.go:101]: Its the number of Istio CRDs created by version 1.1.2. I'll add a variable if any other updates are needed, thanks!

File: examples/kubernetes-istio/istio-sidecar-injector-configmap-debug.yaml [hunk 4]
```diff
@@ -9,7 +9,7 @@ data:
     template: |-
       initContainers:
       - name: istio-init
-        image: docker.io/cilium/istio_proxy_init:0.7.0
+        image: docker.io/cilium/istio_proxy_init:1.1.2
```
Comment (ianvernon) [inline examples/kubernetes-istio/istio-sidecar-injector-configmap-debug.yaml:12]: Why not 1.1.3?

**Pull request merged**